### PR TITLE
Read decomposition file path from reference TaxBenefitSystem with reforms

### PR DESCRIPTION
**Pull request by benjello**

### CODE REVIEW

**Comment by benjello** [comment]
@cbenz @fpagnoux is this good to merge according to your standards ?

**Comment by benjello** [comment]
Need to merge this PR before dealing with #418.

**Comment by benjello** [comment]
@fpagnoux @cbenz : I would like to merge these minor changes to decompositions to avoid accumulating debt ;-)
The version number might badly chosen, please tell me if I should change it.

**Comment by benjello** [comment]
I would like to merge this PR today.

File: openfisca_core/decompositions.py [hunk 1]
```diff
@@ -37,7 +37,11 @@ def calculate(simulations, decomposition_json):
 
 def get_decomposition_json(tax_benefit_system, xml_file_path = None):
     if xml_file_path is None:
-        xml_file_path = os.path.join(tax_benefit_system.DECOMP_DIR, tax_benefit_system.DEFAULT_DECOMP_FILE)
+        try:
```
Comment (fpagnoux) [inline openfisca_core/decompositions.py]: Je ne connais pas exactement le contexte, mais passer par une exception catchée pour faire de la disjonction de cas ce n'est pas trop une bonne pratique.

As-tu vérifié qu'il y avait encore besoin de faire ça depuis que tu as fait l'upstream merge? Si c'est dans le cas d'une réforme, j'ai rajouté il y a peu un `__getatrr__` sur les réformes qui fait que si elles ne trouvent pas un attribut qu'on leur demande, elles vont automatiquement chercher dans le TBS de référence.

> Au passage, auto-feedback par rapport à notre discussion de l'autre jour, c'est une mauvaise pratique de ma part que d'avoir changé le comportement des réformes au milieu d'une PR sur les entités... Un vilain cavalier législatif de développeur ;).

Comment (benjello) [inline openfisca_core/decompositions.py]: Non. Je vais vérifier cela. Merci @fpagnoux.

File: openfisca_core/reforms.py [hunk 2]
```diff
@@ -30,6 +30,7 @@ def __init__(self, reference):
         self._legislation_json = reference.get_legislation()
         self.compact_legislation_by_instant_cache = reference.compact_legislation_by_instant_cache
         self.column_by_name = reference.column_by_name.copy()
+        self.decomposition_file_path = reference.decomposition_file_path
```
Comment (fpagnoux) [inline openfisca_core/reforms.py]: Je ne suis pas sûr que cette ligne soit absolument nécessaire (le `__getattr__` devrait gérer ça tout seul).

Comment (benjello) [inline openfisca_core/reforms.py]: Cela ne marchait pas sans cette ligne.

Comment (fpagnoux) [inline openfisca_core/reforms.py]: Ok !

**Comment by benjello** [comment]
This should be merged with [the companion openfisca-france PR](https://github.com/openfisca/openfisca-france/pull/625)

**Comment by benjello** [comment]
@cbenz : we need your OK on this since it may affect the web-ui etc

**Comment by fpagnoux** [comment]
This branch is rotting ;)

**Comment by cbenz** [comment]
Well done @benjello !

**Comment by fpagnoux** [comment]
Git history before :

image

Git history after this PR and two others :

image

😢

**Comment by cbenz** [comment]
Yes @fpagnoux that's true. But I can assure you that we spent together with @benjello quite 45 min to solve this problem. Not removing all the past merge commits, but only about one.

I think we really should talk about it and find a solution, with @MattiSG! That's not as simple as I thought.

In particular when we merged the branch of the PR into master, on my machine, then pushed the merge-commit in the PR branch, GitHub presented to us a huge diff that was unreadable. If we did the contrary (merging master into the PR branch), the diff was OK.

**Comment by fpagnoux** [comment]
The diff length of this PR is 17 lines. In the worst case, you can checkout master and re-apply manually the changes.

**Comment by fpagnoux** [comment]
Contributors to core manage their time as they can, and it's their responsibility to choose to upstream merge (we've made it clear that is was a bad practise, but this is just a recommandation) or to do clean rebases.

However, when the code enters the `master` codebase, it needs to be clean. Otherwise we have a moral hazard : contributors benefit from going faster, the core team pay the price with poor quality.

**Comment by fpagnoux** [comment]
Even simpler than copy paste : 

- `git checkout` the commit where the code is in the state you want
- `git reset master`

You will then be on `master`, with all the changes between master and your state uncommitted. You can then recommit the way you want.

**Pull request merged**